presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/HistoryTable.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.iceberg;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.prestosql.spi.connector.ColumnMetadata;
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.connector.ConnectorTableMetadata;
+import io.prestosql.spi.connector.ConnectorTransactionHandle;
+import io.prestosql.spi.connector.InMemoryRecordSet;
+import io.prestosql.spi.connector.RecordCursor;
+import io.prestosql.spi.connector.SchemaTableName;
+import io.prestosql.spi.connector.SystemTable;
+import io.prestosql.spi.predicate.TupleDomain;
+import org.apache.iceberg.HistoryEntry;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.util.SnapshotUtil;
+
+import java.util.List;
+import java.util.Set;
+
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.spi.type.BooleanType.BOOLEAN;
+import static io.prestosql.spi.type.TimestampType.TIMESTAMP;
+import static java.util.Objects.requireNonNull;
+
+public class HistoryTable
+        implements SystemTable
+{
+    private final SchemaTableName tableName;
+    private final Table icebergTable;
+
+    private static final List<ColumnMetadata> COLUMNS = new ImmutableList.Builder<ColumnMetadata>().add(
```

Comment:
Use `ImmutableList.builder()` and a separate `.add()` call for each item

Reply:
I don't follow why usage of `ImmutableList.builder()` with *separate* `.add()` calls are preferable here. Would not `ImmutableList.of(...)` provide less verbose construct? If `of(...)` should be avoided, what is wrong with the existing code?

Reply:
@vrozov I thought `ImmutableList.of(...)` can only take a limited number of parameters. Looks like I was wrong. So why do we prefer `ImmutableList.builder()` to `ImmutableList.of(...)` in this case? @electrum 

In addition, I tried `ImmutableList.builder()` before but it doesn't work here because the compiler makes `ImmutableList.builder()` return a `ImmutableList.Builder<Object>` instead of `ImmutableList.Builder<ColumnMetadata>`.

Reply:
Sorry, I should have been more explicit when I suggested using a builder. We want the code to look like this:
```java
ImmutableList.<ColumnMetadata>builder()
        .add(...)
        .add(...)
        .build();
```
This is purely for syntactic reasons. It avoids special casing the last item, which leads to cleaner diffs when you add/remove items in the future.

Reply:
Oh I see, thanks.